### PR TITLE
Allow availability items with only deprecated versions.

### PR DESCRIPTION
**Pull request by sofiaromorales**
Bug/issue #, if applicable: 138034711

## Summary

Symbols in the symbol graph can specify a deprecated availability version without an introduced version. Currently, DocC excludes these availability items entirely. This change allows that items without an introduced version but with a deprecated version are retained instead of being filtered out.

This adjustment is necessary because the `@available` attribute allows defining only a deprecated version without an introduced version, a scenario that is not currently reflected in the final documentation.

## Dependencies

_Describe any dependencies this PR might have, such as an associated branch in another repository._

## Testing

[Availability.docc.zip](https://github.com/user-attachments/files/17547654/Availability.docc.zip)

Steps:
1.Preview the attached doc catalog.
2.Assert that the platform iOS is showing as deprecated in the symbol Foo.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [X] Added tests
- [X] Ran the `./bin/test` script and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

File: Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift [hunk 1]
```diff
@@ -1253,12 +1253,6 @@ public struct RenderNodeTranslator: SemanticVisitor {
             
             return availability.availability
                 .compactMap { availability -> AvailabilityRenderItem? in
-                    // Filter items with insufficient availability data.
-                    // Allow availability without version information, but only if
-                    // both, introduced and deprecated, are nil.
-                    if availability.introducedVersion == nil && availability.deprecatedVersion != nil {
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift:1259]: Question: is it possible to write `@available` attributes without either an introduced or deprecated version. If so, what do we want to do with such availability items? Do we want to display them or filter them out?

In other words, should we remove this check completely (to display such availability items) or should we update the check to see that some version information is non-nil:
```swift
if availability.introducedVersion == nil, 
   availability.deprecatedVersion == nil, 
   availability.obsoletedVersion  == nil
{
   return nil
}
```

**Review by d-ronnqvist — APPROVED** [review]
This looks good to me. Thanks for the added tests.

I only have one non-blocking question regarding what's the expected behavior for availability items without either an introduced or deprecated version.

File: Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift (hunk 1 above)
Comment (sofiaromorales) [reply Sources/SwiftDocC/Model/Rendering/RenderNodeTranslator.swift:1259]: Good catch, the `@available` attribute accepts the following values:
`Expected 'available' option such as 'unavailable', 'introduced', 'deprecated', 'obsoleted', 'message', or 'renamed'`.

`unavailable` `introduced` and `deprecated` are correctly handled by DocC.
`message` and `renamed` don't surface the symbol graph.

The only case left it's `obsoleted`, this is not something we represent in the documentation today so in this commit `https://github.com/swiftlang/swift-docc/pull/1079/commits/d07cf9a0c7ef7eb1305da3acfca1adeb858f8936` I'm filtering it out.

**Comment by sofiaromorales** [comment]
@swift-ci please test

**Comment by sofiaromorales** [comment]
@swift-ci please test

**Review by franklinsch — APPROVED** [review]

**Comment by sofiaromorales** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**